### PR TITLE
BUG: ArrowDtype.construct_from_string round-trip

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by mroeschke** [comment]
Might make sense to feature request this to Arrow to support parsing tz's in `type_for_alias` so we don't have to roll our own?

**Comment by jbrockmendel** [comment]
> Might make sense to feature request this to Arrow to support parsing tz's in type_for_alias so we don't have to roll our own?

I'll do that separately, sure.

**Comment by jbrockmendel** [comment]
issue has been opened upstream + green

**Comment by jbrockmendel** [comment]
@mroeschke gentle ping on pyarrow-xfail PRs

File: pandas/core/arrays/arrow/dtype.py [hunk 1]
```diff
@@ -157,6 +165,37 @@ def construct_from_string(cls, string: str) -> ArrowDtype:
             raise TypeError(f"'{base_type}' is not a valid pyarrow data type.") from err
         return cls(pa_dtype)
 
+    @classmethod
+    def _parse_temporal_dtype_string(cls, string: str) -> ArrowDtype:
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/dtype.py:169]: Could you add a comment that this isn't needed once https://github.com/apache/arrow/issues/33642 is addressed?

File: pandas/tests/extension/test_arrow.py [hunk 2]
```diff
@@ -1261,8 +1209,9 @@ def test_invalid_other_comp(self, data, comparison_op):
 
 
 def test_arrowdtype_construct_from_string_type_with_unsupported_parameters():
-    with pytest.raises(NotImplementedError, match="Passing pyarrow type"):
-        ArrowDtype.construct_from_string("timestamp[s, tz=UTC][pyarrow]")
```
Comment (mroeschke) [inline pandas/tests/extension/test_arrow.py:1265]: Could we keep this test but with a type (possibly fake) so we still test this exception?

File: pandas/tests/extension/test_arrow.py [hunk 3]
```diff
@@ -1266,7 +1214,12 @@ def test_invalid_other_comp(self, data, comparison_op):
 
 def test_arrowdtype_construct_from_string_type_with_unsupported_parameters():
     with pytest.raises(NotImplementedError, match="Passing pyarrow type"):
-        ArrowDtype.construct_from_string("timestamp[s, tz=UTC][pyarrow]")
+        ArrowDtype.construct_from_string("not_a_real_dype[s, tz=UTC][pyarrow]")
+
+    # but as of GH#50689, timestamptz is supported
```
Comment (mroeschke) [inline pandas/tests/extension/test_arrow.py:1219]: In a follow up could you make this it's own test? (Nit: doesn't really fit the name of the test"

**Review by mroeschke — APPROVED** [review]
LGTM pending greenish

File: pandas/tests/extension/test_arrow.py (hunk 3 above)
Comment (jbrockmendel) [reply pandas/tests/extension/test_arrow.py:1219]: fair enough, will do

**Comment by jbrockmendel** [comment]
CI timeouts unrelated, merging.

**Pull request merged**